src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +158,1547 @@ ActorInit En_Jso_InitVars = {
+        this->actor.velocity.y = 0.0f;
+        this->actor.speed = 0.0f;
+        this->timer = 0;
+        sIsAttacking = false;
+        this->isAttacking = false;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        this->attackWaitTimer = Rand_S16Offset(10, 10);
+        EnJso_SetupCirclePlayer(this, play);
+    }
+}
+
+void EnJso_SetupDead(EnJso* this, PlayState* play) {
+    AudioSfx_SetChannelIO(&this->actor.projectedPos, NA_SE_EN_ANSATSUSYA_DASH_2, 0);
+    EnJso_ChangeAnim(this, EN_JSO_ANIM_DAMAGED);
+
+    if (((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
+         (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) &&
+        (this->drawDmgEffAlpha == 0)) {
+        this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+    }
+
+    this->actor.flags &= ~(ACTOR_FLAG_TARGETABLE | ACTOR_FLAG_UNFRIENDLY);
+    this->actor.flags |= ACTOR_FLAG_CANT_LOCK_ON;
+    this->actor.speed = 0.0f;
+    this->disableBlure = true;
+    this->timer = 30;
+    Enemy_StartFinishingBlow(play, &this->actor);
+    Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_DEAD);
+    this->action = EN_JSO_ACTION_DEAD;
+    this->actionFunc = EnJso_Dead;
+}
+
+/**
+ * Plays the damaged animation for 30 frames, then the Garo falls down and starts talking to the player.
+ */
+void EnJso_Dead(EnJso* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+
+    if ((this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_SFX) ||
+        (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX)) {
+        if (this->drawDmgEffAlpha != 0) {
+            Actor_SpawnIceEffects(play, &this->actor, this->bodyPartsPos, EN_JSO_BODYPART_MAX, 2, 0.3f, 0.2f);
+            this->drawDmgEffAlpha = 0;
+            this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FIRE;
+        } else {
+            return;
+        }
+    }
+
+    if ((curFrame >= this->animEndFrame) && (this->timer == 0)) {
+        EnJso_SetupFallDownAndTalk(this, play);
+    }
+}
+
+void EnJso_SetupFallDownAndTalk(EnJso* this, PlayState* play) {
+    EnJso_ChangeAnim(this, EN_JSO_ANIM_FALL_DOWN);
+    this->textIndex = 2;
+    this->actor.colChkInfo.mass = 255;
+    this->actor.flags |= (ACTOR_FLAG_TARGETABLE | ACTOR_FLAG_FRIENDLY);
+    func_800BC154(play, &play->actorCtx, &this->actor, ACTORCAT_NPC);
+    this->actor.flags &= ~ACTOR_FLAG_CANT_LOCK_ON;
+    this->actor.flags &= ~ACTOR_FLAG_100000;
+    this->action = EN_JSO_ACTION_FALL_DOWN_AND_TALK;
+    this->actionFunc = EnJso_FallDownAndTalk;
+}
+
+/**
+ * Handles the first part of the dialogue the Garo gives when defeated.
+ */
+void EnJso_FallDownAndTalk(EnJso* this, PlayState* play) {
+    f32 curFrame = this->skelAnime.curFrame;
+    Player* player = GET_PLAYER(play);
+
+    if (!(curFrame < this->animEndFrame) && !Play_InCsMode(play) && (play->msgCtx.msgLength == 0)) {
+        if (fabsf(player->actor.world.pos.y - this->actor.world.pos.y) < 60.0f) {
+            Player* player2 = GET_PLAYER(play);
```

Comment:
Do you think this `player2` could be a fake match? It appears a couple times in this actor, maybe it was just the style of the dev?

Reply:
One thing we can say for certain is that calling `GET_PLAYER` a second time in this if-block is almost assuredly real. Here's the function you highlighted with the `player2` stuff entirely removed. Notice the missing `lw v0,0x1ccc(a3)` in the right column; there are two in the original, but only one in the current:
![image](https://github.com/zeldaret/mm/assets/12245827/6440ec60-ff13-40d7-b20d-e2fe2faac98d)

Anytime you see a `player2` in this file (which is only twice, mind you), the key thing to note is that the second `GET_PLAYER` macro is probably real, because you have to do it twice in the function to get that `lw v0,0x1ccc(a3)` to show up twice. Now, could you potentially match this by doing `GET_PLAYER` twice but only using the first `player`? Maybe, but I've never gotten it to match like that, and it doesn't feel worth it because we have to call `GET_PLAYER` twice anyways, so we might as well use it.

So in short, I think this is actually real and just a weird stylistic quirk. If you can figure out how to get that second load to show up with just a single `GET_PLAYER`, I'd be fully willing to look into this further. But from what I can tell, we need two `GET_PLAYER`s here.

Reply:
Yeah, that seems like enough of an argument for me, thanks for looking into it